stl/src/filesystem.cpp

```diff
@@ -657,11 +657,15 @@ __std_win_error __stdcall __std_fs_get_file_id(__std_fs_file_id* const _Id, cons
     }
 
     _Last_error = __std_win_error{GetLastError()};
-    if (_Last_error != __std_win_error::_Invalid_parameter && _Last_error != __std_win_error::_Invalid_function) {
+    switch (_Last_error) {
+    case __std_win_error::_Invalid_parameter: // Older Windows versions
+    case __std_win_error::_Invalid_function: // Windows 10 1607
+    case __std_win_error::_Not_supported: // Current Windows versions on ReFS (DevCom-857535)

```

Comment:
"Current" here could become invalid if this behavior changes in Windows. Maybe just `// POSIX delete not supported (e.g., DevCom-857535)`?